cabalscan/src/CabalScan/RuleGenerator.hs

```diff
@@ -281,6 +284,20 @@ depPackageNames = concatMap depNames . Cabal.targetBuildDepends
         in
           map identifierOf $ Set.toList $ Cabal.depLibraries dep
 
+data FoundModulePath = FoundModulePath
```

Comment:
This type is requiring a module to be put together with its boot module. Is this necessary for any further processing? As far as I can see this link is discarded in `foundModulePathToPathList` almost immediately after going through the trouble of putting it together.

Reply:
Seems that boot files are ignored if no corresponding module is present, but this is more behavior than has been discussed so far. Is this why you want to return either a 2-tuple or a `FoundModulePath`? I still don't get it :)

Reply:
Yep, if I only return the first module I match with an extension, either only the `.hs` file or only the `.hs-boot` file will match, but not both. So we need to return both of them. Either a tuple, or a list, or a record. I prefer the record approach here, more clear than a tuple.

Reply:
Thanks for the elaboration. That starts to make sense.

Now, note that boot files can end with extension `.lhs-boot` as well. And I'm realizing that we would need to define the behavior when both `.lhs-boot` and `.hs-boot` files are present. And similarly in older code, when ".hs" and ".lhs" files are both present.

In light of this, how about we just collect all files matching any of the extensions ".hs", ".lhs", ".hs-boot", ".lhs-boot", and ".hsc"?

We could leave it to `rules_haskell` to fail if any of file combinations is incompatible.

Reply:
Good point, thanks.

Reply:
Done